.github/workflows/checkpatch.yml

```diff
@@ -0,0 +1,51 @@
+---
+# Tools that can save round-trips to github and a lot of time:
+#
+# yamllint -f parsable pull_request.yml
+# pip3 install ruamel.yaml.cmd
+# yaml merge-expand pull_request.yml exp.yml &&
+#    diff -w -u pull_request.yml exp.yml
+#
+# github.com also has a powerful web editor that can be used without
+# committing.
+
+# This is the name of this workflow and should technically be called
+# something like "Main Workflow" but the place where most people see
+# this name is the Checks window next to other, non-github checks.
+name: checkpatch
+
+# yamllint disable-line rule:truthy
+on: pull_request
+
+jobs:
+  checkpatch:
+    runs-on: ubuntu-latest
+    # skip upstream merge PR
+    if: ${{ !contains(github.head_ref, 'merge/sound-upstream') }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: set up env for checkpath
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y wget jq codespell
+
+      - name: check patch
+        run: |
+          RET=0
+          PR=${GITHUB_REF#"refs/pull/"}
+          PRNUM=${PR%"/merge"}
+          # get GitHub PR commits API URL
+          URL="${GITHUB_API_URL}/repos/${GITHUB_REPOSITORY}/pulls/${PRNUM}/commits"
+          patchlist=$(curl $URL -X GET -s | jq '.[].html_url' -r)
+          # checkpatch.pl on each patch
+          echo
+          for patchurl in $patchlist; do
+              sha=${patchurl##*/}
+              wget -q "${patchurl}.patch" -O "${sha}.patch"
```

Comment:
This is incredibly complicated. `.travis.yml` does all this with one line, did you not look at it?

Reply:
Got it: this is a `fetch-depth: 1` clone by default. Rather than fetching the patches one by one with wget, you can fetch them all at once (and only them) with git:
```shell
capped_len=$((patchlistlen < 500 ? patchlistlen : 500))
git fetch --depth $capped_len origin
```
Then use `checkpatch -g HEAD~$capped_len` normally.

Reply:
The main propose for these wget is to avoid git checkout at all. But it seems I will need the checkpatch.pl from our repo.

I will remove the checkpatch part as I think I may not understand what we are doing in trvais.yml and how could we avoid the long time git check out.